src/main/java/org/opensearch/security/auth/UserInjector.java

```diff
@@ -93,14 +93,16 @@ public TransportAddress getTransportAddress() {
         }
 
         public void setTransportAddress(String addr) throws UnknownHostException, IllegalArgumentException {
-            // format is ip:port
-            String[] ipAndPort = addr.split(":");
-            if (ipAndPort.length != 2) {
+            int lastColonIndex = addr.lastIndexOf(':');
+            if (lastColonIndex == -1) {
```

Comment:
Side question: are we always guaranteed to get a port with the IP address? If not then this assumption/check would probably fail. However, this is consistent with the behavior today so it is probably fine.

Reply:
Can we add a more robust check than relying on last index of colon since IPv6 addresses also contain colon in them. My suggestion here would to use regex-matching to allow for ports to be read correctly if present and not if not present.

Reply:
My thinking behind doing it this way is that we do not need to re-invent the wheel. Since the library is already properly parsing an IP in either v4 or v6 format, all we need to do is to extract the supposed port and assume the rest of the string is the address. If it is malformed in any way the library would catch this, assuming that the port is always present (which the existing code does). Does that make sense?

Reply:
This looks fine to me and User Injection will always have port. See comment [here](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/auth/UserInjector.java#L128). If the port doesn't exist, throw an IllegalArgumentException. The call to [Integer.parseInt](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/auth/UserInjector.java#L103) will also ensure that its a valid integer value. You could get even more restrictive and check that its between 1–65535, but that is probably overkill.